### PR TITLE
Improved private registry docs

### PR DESCRIPTION
**Pull request by jbayer**
Based on [this report in the forum](https://discuss.hashicorp.com/t/if-you-look-for-private-docker-usage/16175), I created a PR to improve the private registry documentation.

### CODE REVIEW

File: website/content/docs/lifecycle/build.mdx [hunk 1]
```diff
@@ -232,54 +232,71 @@ the build/registry phases are one and the same.
 ### Private Registries
 
 When using the docker plugin to push images to private container registries, the
-registry authentication is configured "out of band" from Waypoint. For example
-your `waypoint.hcl` may have a `build` stanza with a `registry` section pointing
-to a private registry:
+registry authentication may either be configured explicitly or "out of band"
+from Waypoint.
+
+_Explicit Authentication_
```
Comment (mitchellh) [inline website/content/docs/lifecycle/build.mdx:238]: You can use `h4`s for this if you want: `####`

File: website/content/docs/lifecycle/deploy.mdx [hunk 2]
```diff
@@ -78,3 +78,45 @@ $ waypoint deploy -release=false
...
 -> A future Waypoint version will allow this to be disabled in the project
 configuration file.
+
+## Private Registries
+
+Deployment platforms that pull an app container image
+from a private registry need configuration to authenticate
+to a private registry.
+
+_Kubernetes with a Private Registry_
```
Comment (mitchellh) [inline website/content/docs/lifecycle/deploy.mdx:88]: h3 here if you want.

File: website/content/docs/lifecycle/build.mdx (hunk 1 above)
Comment (jbayer) [reply website/content/docs/lifecycle/build.mdx]: Thanks, addressed this.

File: website/content/docs/lifecycle/deploy.mdx (hunk 2 above)
Comment (jbayer) [reply website/content/docs/lifecycle/deploy.mdx]: Thanks, addressed this.

**Pull request merged**